### PR TITLE
michelgotta/profile-page

### PR DESCRIPTION
**Pull request by michelgotta**
First WIP pull request for testing / review
– Tips and activities missing
– Todos: Header stats data hard wired dummy data

### CODE REVIEW

File: app/assets/v2/css/base.css [hunk 1]
```diff
@@ -896,191 +896,170 @@ input[type=text].loading {
...
+.profile-header {
 
-.profile_details li:nth-child(2n+0) .dot {
-  background: #919191;
 }
 
-.profile_details li:nth-child(2n+1) .dot {
-  background: black;
+.profile-header > .container {
+  padding-top: 50px;
```
Comment (thelostone-mc) [inline app/assets/v2/css/base.css]: @michelgotta could we move from `px` -> `rem` ?

**Comment by thelostone-mc** [comment]
@michelgotta could we move from `px` -> `rem` in CSS?

File: app/assets/v2/css/base.css [hunk 2]
```diff
@@ -896,191 +896,170 @@ input[type=text].loading {
...
-.profile_details .dot {
-  border-radius: 6px;
-  width: 12px;
-  height: 12px;
-  margin-left: -55px;
-  float: left;
-  vertical-align: middle;
-  margin-top: 20px;
-}
+/* Profile details */
```
Comment (thelostone-mc) [inline app/assets/v2/css/base.css]: Could we move this into it's own css file ?

**Comment by michelgotta** [comment]
@thelostone-mc Thanks for the first review. Changed the px to rem in the css and moved it to a separate file for now.

Added the tips to the profile as well. The hard wired dummy data should be changed by @mbeacom or @owocki . After that, feel free to merge my branch into the master.

**Comment by michelgotta** [comment]
@owocki Latest commit is now ready for review. Stats and Activities data hard wired in views.py.

**Comment by mbeacom** [comment]
@michelgotta Checking this out now.  Thanks!

File: app/assets/v2/images/check-green.svg [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (mbeacom) [inline app/assets/v2/images/check-green.svg]: Squashing this

File: app/dashboard/templates/profile_details.html [hunk 4]
```diff
@@ -3,238 +3,491 @@
...
+            {% for group in activities %}
+              <div class="row">
+                <div class="col-12">
+                  <p><small>{{group.title}}</small></p>
+                </div>
+                {% if group.completed %}
+                  <div class="container">
+                    <div class="row profile-bounties__activities-group mb-3 p-2">
+                      <div class="col-12">
+                        <div class="profile-bounties__group-title profile-bounties__group-title--completed">Completed {{group.completed|length}} {% if group.completed|length > 1 %}Bounties{% else %}Bounty{% endif %} <img src="/static/v2/images/check-green.svg"></div>
```
Comment (mbeacom) [inline app/dashboard/templates/profile_details.html]: We should use `{% static "v2/images/check-green.svg" %}` here

Comment (mbeacom) [inline app/dashboard/templates/profile_details.html]: We should use `{% static "v2/images/check-green.svg" %}` here

File: app/dashboard/templates/profile_details.html [hunk 5]
```diff
@@ -3,238 +3,491 @@
...
+
+                        {% for bounty in group.completed %}
+                          <div class="row result bounty_row {{bounty.network}}" {% if bounty.github_url %}href="{{bounty.github_url}}"{% endif %}>
+                            <div class="avatar-container col-1 justify-content-center special_tag hide_min_viewport">
+                              <img class="avatar" src="/funding/avatar?repo=https://github.com/{{bounty.bounty_owner_github_username}}"/>
```
Comment (mbeacom) [inline app/dashboard/templates/profile_details.html]: Can we switch this to use `{% url "" %}` for internal URL references throughout this file?

File: app/dashboard/views.py [hunk 6]
```diff
@@ -837,7 +837,28 @@ def profile(request, handle):
         'stats': profile.stats,
         'bounties': profile.bounties,
         'tips': Tip.objects.filter(username=handle, network='mainnet'),
+        # TODO: Fill following with real data
```
Comment (mbeacom) [inline app/dashboard/views.py]: 👍 I'll tackle this for you once I'm back from pycon this evening.

**Comment by owocki** [comment]
my notes:

0. this looks really good!
1. 'bounties completed' doesnt have a number next to it.  should be `0` if we can't find any.
2. 'no tips found' has a different style than the other 'no results' pages
3. what is "# funder" and "# contributor" in the top right?
4. [for results that don't have an avatar can we put an empty avatar in the rows?](http://bits.owocki.com/2o163H180Z1z/Screen%20Shot%202018-05-16%20at%2011.00.25%20AM.png)
5. responsive design looks good!

**Comment by owocki** [comment]
cc @mbeacom for if he has comments too

**Comment by michelgotta** [comment]
@owocki

0. Thanks! It's fun to work on this project
1. Should be easy to fix
2. I will check the style for this blank slate
3. #1 funder / contributor should be the users position/rank on the leaderboard
4. I will check this issue
5. Thanks again

**Comment by michelgotta** [comment]
Updated this PR with

@thelostone-mc 
- removed blocks with no results

@owocki 
> for results that don't have an avatar can we put an empty avatar in the rows?
Do we have a default avatar example anywhere?

> 'bounties completed' doesnt have a number next to it. should be 0 if we can't find any.
I think I found the right data for this in the model. Added this to the view. @mbeacom will fill the missing pieces in the "stats" section.

**Comment by owocki** [comment]
@michelgotta you can use this image for the blank avatars http://bits.owocki.com/0j361H222B2s/gitcoin.png

**Comment by thelostone-mc** [comment]
Pulled up the code and checked it out ^_^ Everything looks neat 

Just a few changes though: 

##### General

@PixelantDesign is this is color you wanted on hove
( we haven't used this anywhere in the site, looks off )

x1

##### Tablet

Could we increase the width for both the boxes, so that the contributor & Funder box  remains same height ?

screen shot 2018-05-17 at 2 45 04 pm

Bounty Details seems to be overflowing the boundary 

screenshot-2018-5-17 owocki gitcoin

##### @720px  

The image and bounty description are broken into two lines -> could we fix that up ? 

screenshot-2018-5-17 owocki gitcoin 1

##### Mobile
This would look nicer if it was center aligned as
```
Photo 
name 
social links
 ```

screen shot 2018-05-17 at 2 44 00 pm

File: app/dashboard/views.py (hunk 6 above)
Comment (mbeacom) [reply app/dashboard/views.py]: Sorry for the delay here.  I'm adding this to my list for tomorrow.
Thanks!

**Comment by michelgotta** [comment]
Thanks everyone for the reviews and the great feedback. Just pushed some more changes to the pull request.

@thelostone-mc 
**"Show all" Button hover state:**
Added a more subtle styling for the hover of the button. There is no constant behavior throughout the app for button styling. Mentioned it here: https://github.com/gitcoinco/web/issues/1173

**Bounty Details seems to be overflowing** and **Avatar and description in two lines** on tablet breakpoint
This is the original behavior and reused code from the issue explorer. Long term: There should be a  framework in use where we can reuse these template bits like React. Won't change things here. Should be changed in explorer first.

**contributor & Funder box text breaking** on tablet
Decreased the font size. Better than making the boxes more wide if we have a long username or github handle on the left

**center aligned info in header** on mobile/tablet
Changed this, looks great and is more flexible with long usernames / github handle etc.

Info: @mbeacom will provide some additional changes for real "Recent Activity" data from the database.

**Comment by thelostone-mc** [comment]
@michelgotta checked out the code ! 
Everything looks good to me ! The overflow table bit -> i'll fix it up once this gets merged in a generic manner

**Comment by mbeacom** [comment]
General changes for the BE are in.  I'm going to update the tests and add some optimizations to the new queries, but we should be pretty close!

**Comment by owocki** [comment]
testing now!

**Comment by owocki** [comment]
when i run migrations


```
  Applying dashboard.0076_auto_20180520_0357...Exception in dashboard migration 0076 - (Profile matching query does not exist.) - Bounty object (1462)
Exception in dashboard migration 0076 - (Profile matching query does not exist.) - Bounty object (535)
Exception in dashboard migration 0076 - (Profile matching query does not exist.) - Bounty object (429)
Exception in dashboard migration 0076 - (Profile matching query does not exist.) - Bounty object (1175)
Exception in dashboard migration 0076 - (Profile matching query does not exist.) - Bounty object (2688)
Exception in dashboard migration 0076 - (Profile matching query does not exist.) - Bounty object (1116)
Exception in dashboard migration 0076 - (Profile matching query does not exist.) - Bounty object (2739)
Exception in dashboard migration 0076 - (Profile matching query does not exist.) - Bounty object (2073)
Exception in dashboard migration 0076 - (Profile matching query does not exist.) - Bounty object (2362)
Exception in dashboard migration 0076 - (Profile matching query does not exist.) - Bounty object (2918)
```
(goes on like this for pages)

**Comment by owocki** [comment]
Then when I try to access a profile on the branch
```
django.core.exceptions.FieldError: Cannot resolve keyword 'key' into field. Choices are: active, amount, created_on, github_username, id, leaderboard, modified_on
```

and yes I've run my migrations

**Comment by owocki** [comment]
frontend issues

1. the 'load more' / 'show more' button doesnt seem to show up anymore.  now it just shows 100s of bounties on my profile when i first nav there.
2. for [bounties that dont have an avatar](http://bits.owocki.com/1D0Y0U0K1C2e/Screen%20Shot%202018-05-22%20at%205.36.54%20PM.png) it'd be great to show _something_ here instead of just a blank square
3. [latest activities](http://bits.owocki.com/3X1e33003A1L/Screen%20Shot%202018-05-22%20at%205.37.48%20PM.png) are sorted by date.. but none of the rows in the table say when they happened.  if this is gonna be a 'feed' shouldnt it show the relative time in each row?

otherwise this is looking pretty good!

**Comment by owocki** [comment]
cc @mbeacom on the backend issues

**Comment by michelgotta** [comment]
@owocki I will check for the Avatar. Shouldn't be the default generated GitHub avatar displayed when a user has no uploaded one? If we can stage a running version this it would be easier to check this.

**Comment by owocki** [comment]
> @owocki I will check for the Avatar. Shouldn't be the default generated GitHub avatar displayed when a user has no uploaded one? If we can stage a running version this it would be easier to check this.

547cd56c28d2c39443e3a233e2fd3c8bb5d74fd4 fixes this

**Comment by owocki** [comment]
show more link is working now

**Comment by owocki** [comment]
i think this is ready to ship... @mbeacom @PixelantDesign what do you think?

**Comment by thelostone-mc** [comment]
@michelgotta it's up on staging here http://stage.gitcoin.co/

@owocki 
me and @mbeacom had a few concerns regarding the number of APIs being fired but we can work upon it post merge too !

**Comment by PixelantDesign** [comment]
looks good to me! let's push!

**Comment by michelgotta** [comment]
@thelostone-mc I am not really satisfied with the mix of async API calls and some data coming from the python view controller. One objective was `Try and reuse as much of it as possible` from the explorer, but some data I couldn't aggregate through the API. Long term goal should be a rebuild on a frame work like React or Next.js.

Two improvements to easily reduce the payload:
- Activate GZIP for API data (would reduce data up to 80%, Test: ~130Kb --> ~30Kb)
- Don't deliver whole description for bounties on "list" calls (or make it optional like `&description=false`)
@mbeacom @owocki

**Comment by michelgotta** [comment]
@mbeacom Could you provide help with the conflict in the models.py?

**Comment by mbeacom** [comment]
@michelgotta I'll take care of it in a moment! Thanks for the heads up.

**Comment by mbeacom** [comment]
Resolved.  Regarding the API situation, gzip would make sense.  That said, if we're already querying the vast majority of this data and providing it via the v/c, why not use the data already available versus making another call to grab it from the API?  That or why perform any of it on the BE? 🤔

**Comment by michelgotta** [comment]
@thelostone-mc Just saw this is not in the stage. Is there something missing?

**Comment by thelostone-mc** [comment]
@mbeacom took it off staging cause we were pushing other things in !
@mbeacom Wanna redeploy it / we good for merge ?

**Comment by owocki** [comment]
did someone force push here ?  theres a bucnh of new commits on the branch, and i dont see any of my commits on there anymore

File: app/dashboard/views.py (hunk 6 above)
Comment (owocki) [reply app/dashboard/views.py]: is this in? i still see the TODO on the branch

**Comment by michelgotta** [comment]
AHhhhh, I'm so sorry… I think I messed up this up. Is is some kind of merge hell. 😭 I'm trying to fix this…

**Comment by owocki** [comment]
@michelgotta no worries.. it happens..

@mbeacom and i started a new branch called `new_profile` whcih has some of the preserved commits on it..  do you have the commits still pre-merge?  if so, could you push them up somewhere we can take a look at them?

**Comment by owocki** [comment]
the only thing i can see that needs to be added to https://github.com/gitcoinco/web/pull/1268 is the 'load more' buttons.. everything else i think i see there.. @michelgotta can you help?

**Comment by owocki** [comment]
closing this in favor of https://github.com/gitcoinco/web/pull/1268

**Comment by owocki** [comment]
heres a v2 ticket -- anything we should add to it? https://github.com/gitcoinco/web/issues/1273

**Pull request closed without merging**